### PR TITLE
Gjoranv/new metrics proxy fixes 

### PR DESCRIPTION
**Pull request by gjoranv**

### CODE REVIEW

**Review by hmusum — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/admin/metricsproxy/MetricsProxyContainerCluster.java [hunk 1]
```diff
@@ -150,19 +147,16 @@ private MetricSet getAdditionalDefaultMetrics() {
...
         Monitoring monitoring = getMonitoringService();
-        if (monitoring != null && ! monitoring.getClustername().equals(""))
-            return monitoring.getClustername();
-        return DEFAULT_NAME_IN_MONITORING_SYSTEM;
+        return monitoring != null && ! monitoring.getClustername().equals("") ?
+                Optional.of(monitoring.getClustername()) : Optional.empty();
     }
 
-    private int getIntervalMinutes() {
+    private Optional<Integer> getIntervalMinutes() {
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/metricsproxy/MetricsProxyContainerCluster.java]: Consider using `OptionalInt`

**Review by bjorncs — APPROVED** [review]
👍

**Review by bjorncs — APPROVED** [review]

**Review by hmusum — APPROVED** [review]

File: metrics-proxy/pom.xml [hunk 2]
```diff
@@ -64,6 +64,13 @@
       <version>${project.version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <!-- Not directly used in this module, but needed to get Import-Packages for JDK packages it exports. -->
```
Comment (bjorncs) [inline metrics-proxy/pom.xml]: Is jdisc_core always required as provided dependency for bundles?

**Review by bjorncs — APPROVED** [review]

File: metrics-proxy/pom.xml (hunk 2 above)
Comment (gjoranv) [reply metrics-proxy/pom.xml]: It exports 'javax.*' packages and so on, so I would in general say 'yes'.

Comment (bjorncs) [reply metrics-proxy/pom.xml]: OK, so it is not necessary for java.* packages.

Comment (gjoranv) [reply metrics-proxy/pom.xml]: No, because OSGi (at least in r7) still allows not importing them.

**Pull request merged**